File: app/src/main/java/com/olbigames/finddifferencesgames/game/Finger.java
package com.olbigames.finddifferencesgames.game;

import android.graphics.Point;

public class Finger {
    public int ID;			// »дентификатор пальца
    public Point Now;
    public Point Before;
    public long wasDown;
    boolean enabled = false;		// Ѕыло ли уже сделано движение
    
    public Finger(int id, int x, int y){
        wasDown = System.currentTimeMillis();
        ID = id;
        Now = Before = new Point(x, y);
    }
    
    public void setNow(int x, int y){
        if(!enabled){
            enabled = true;
            Now = Before = new Point(x, y);
        }else{
            Before = Now;
            Now = new Point(x, y);
        }
    }
}

File: app/src/main/java/com/olbigames/finddifferencesgames/game/Finger.kt
package com.olbigames.finddifferencesgames.game

import android.graphics.Point

class Finger(id: Int, x: Int, y: Int) {
    // Finger id
    var ID: Int
    var pointNow: Point
    var pointBefore: Point
    var wasDown: Long = System.currentTimeMillis()
    // Has the movement been already made
    var enabled = false

    init {
        ID = id
        pointBefore = Point(x, y)
        pointNow = pointBefore
    }

    fun setNow(x: Int, y: Int) {
        if (!enabled) {
            enabled = true
            pointBefore = Point(x, y)
            pointNow = pointBefore
        } else {
            pointBefore = pointNow
            pointNow = Point(x, y)
        }
    }
}